### PR TITLE
Forbid to delete inbox and forbid to rename mailbox to the existing name

### PR DESCRIPTION
**Pull request by filtsin**
https://github.com/greenmail-mail-test/greenmail/issues/399

### CODE REVIEW

File: greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/DeleteCommand.java [hunk 1]
```diff
@@ -41,6 +41,11 @@ protected void doProcess(ImapRequestLineReader request,
                 folder.getFullName().equals(session.getSelected().getFullName())) {
             session.deselect();
         }
+
+        if (mailboxName.toLowerCase().equals("inbox")) {
```
Comment (apinske) [inline greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/DeleteCommand.java:45]: I suggest moving this to `ImapHostManagerImpl` too. You might reuse the check for the INBOX folder as in `renameMailbox`.

File: greenmail-core/src/test/java/com/icegreen/greenmail/imap/commands/ImapProtocolTest.java [hunk 2]
```diff
@@ -335,6 +335,52 @@ public void testRenameFolder() throws MessagingException {
...
     }
 
+    @Test
+    public void testDeleteInbox() throws MessagingException {
+        store.connect("foo@localhost", "pwd");
+        try {
+            IMAPFolder folder = (IMAPFolder) store.getFolder("INBOX");
+            folder.open(Folder.READ_WRITE);
+
+            Response[] ret = (Response[]) folder.doCommand(protocol -> protocol.command("DELETE INBOX", null));
```
Comment (apinske) [inline greenmail-core/src/test/java/com/icegreen/greenmail/imap/commands/ImapProtocolTest.java:345]: Any reason why you don't use `folder.delete()` here?

**Review by apinske — CHANGES_REQUESTED** [review]
Thanks for the contribution. Would you mind looking at my comments, please?

File: greenmail-core/src/test/java/com/icegreen/greenmail/imap/commands/ImapProtocolTest.java (hunk 2 above)
Comment (filtsin) [reply greenmail-core/src/test/java/com/icegreen/greenmail/imap/commands/ImapProtocolTest.java:345]: I assumed that in the file "ImapProtocolTest" we test the response to the IMAP command. If you think that it is ok just to call the delete method, how it should look (how can I check for exceptions being thrown)?

Comment (apinske) [reply greenmail-core/src/test/java/com/icegreen/greenmail/imap/commands/ImapProtocolTest.java:345]: Good point!

File: greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/DeleteCommand.java (hunk 1 above)
Comment (filtsin) [reply greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/DeleteCommand.java]: Moved it to `ImapHostManagerImpl`

**Pull request merged**